samples/notebooks/prediction/SDK_Pytorch_Custom_Predict.ipynb

```diff
@@ -0,0 +1,1603 @@
+   "source": [
+    "!cp requirements.txt $USER_SRC_DIR/requirements.txt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "51e149fdec1b"
+   },
+   "source": [
+    "## Build and push container to Artifact Registry"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "3bdb9a7768a5"
+   },
+   "source": [
+    "### Build your custom container\n",
+    "\n",
+    "To build a custom image, a Dockerfile is necessary where you need to implement what the image looks like. However, with the Custom Prediction Routine features, Vertex SDK also generates Dockerfile and build images for you.\n",
+    "\n",
+    "Using `python:3.7` as a base image by default."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "3d3a6b9ed22b"
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "from google.cloud.aiplatform.prediction import LocalModel\n",
+    "\n",
+    "from src_dir_pytorch.predictor import CprPredictor  # Update this path as the variable $USER_SRC_DIR to import the custom predictor.\n",
+    "\n",
+    "local_model = LocalModel.create_cpr_model(\n",
+    "    USER_SRC_DIR,\n",
+    "    f\"{REGION}-docker.pkg.dev/{PROJECT_ID}/{REPOSITORY}/{IMAGE}\",\n",
+    "    predictor=CprPredictor,\n",
+    "    requirements_path=os.path.join(USER_SRC_DIR, \"requirements.txt\"),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "04c988201499"
+   },
+   "source": [
+    "You can check out the serving container spec of the built image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "f1e7d639b9cc"
+   },
+   "outputs": [],
+   "source": [
+    "local_model.get_serving_container_spec()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "8b62ddf1def3"
+   },
+   "source": [
+    "### Store test instances\n",
+    "\n",
+    "To learn more about formatting input instances in JSON, [read the documentation.](https://cloud.google.com/vertex-ai/docs/predictions/online-predictions-custom-models#request-body-details)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "INPUT_FILE = \"instances.json\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "b6605e9e6186"
+   },
+   "outputs": [],
+   "source": [
+    "%%writefile $INPUT_FILE\n",
+    "{\n",
+    "    \"instances\": [\n",
+    "        [6.7, 3.1, 4.7, 1.5],\n",
+    "        [4.6, 3.1, 1.5, 0.2]\n",
+    "    ]\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "147a555f6c93"
+   },
+   "source": [
+    "### Run and test the container locally"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "8phJWBhe7u9u",
+    "tags": []
+   },
+   "source": [
+    "#### Set up credentials\n",
```

Comment:
We can simplify these notebooks a lot by removing the requirement to sign in for local testing. It doesn't really make sense in the first place that users would have to be signed in for testing their container locally. How hard is it to have our system handle a model artifact that could either be on GCS or in a local directory?

Reply:
I think there are two parts for this.
1. Our local run allows users to pass a local directory.
2. Users write two ways to load models from GCS and local files.
1 is not hard but in my opinion it doesn't make much sense for users to write two ways to load models in order to test images locally. Any it is possible (at a lower chance though) that images work well with local directories but are broken with GCS path if the code in `load` function is wrong.

Reply:
OK, sounds like we should make it easier for the users, then. Let's provide an importable function that can load from either GCS or a local directory, like tf.io.gfile (except we don't want to include all of tensorflow by default)

Reply:
Ack.